src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3566,10 +3567,19 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
                     subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
                     namespace='Microsoft.Compute', type='snapshots', name=s)
     source = GalleryArtifactSource(managed_image=ManagedArtifact(id=managed_image))
-    profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest, end_of_life_date=end_of_life_date,
-                                            target_regions=target_regions or [TargetRegion(name=location)],
-                                            source=source, replica_count=replica_count,
-                                            storage_account_type=storage_account_type)
+    if replication_mode is not None:
+        profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
+                                                end_of_life_date=end_of_life_date,
+                                                target_regions=target_regions or [TargetRegion(name=location)],
+                                                source=source, replica_count=replica_count,
+                                                storage_account_type=storage_account_type,
+                                                replication_mode=replication_mode)
+    else:
+        profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
+                                                end_of_life_date=end_of_life_date,
+                                                target_regions=target_regions or [TargetRegion(name=location)],
+                                                source=source, replica_count=replica_count,
+                                                storage_account_type=storage_account_type)
```

Comment:
May I ask why is the parameter `replication_mode` not allowed to be passed into `None` for `ImageVersionPublishingProfile`?

Reply:
If the parameter passed in `sreplication_mode` is `None`, the return result is `null`, which is the same as the result without passing in parameters.

Reply:
How about
```Python
profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
                                                end_of_life_date=end_of_life_date,
                                                target_regions=target_regions or [TargetRegion(name=location)],
                                                source=source, replica_count=replica_count,
                                                storage_account_type=storage_account_type)
 if replication_mode is not None:
      profile.replication_mode = replication_mode 
```